### PR TITLE
Dev/ej

### PR DESCRIPTION
**Pull request by jelavice**

### CODE REVIEW

**Comment by jelavice** [comment]
Added dox for the localization functionality.

**Comment by nubertj** [comment]
this is great, I will test it this weekend

**Comment by tutunarsl** [comment]
Will review and test as soon as possible

**Comment by jelavice** [comment]
thanks for reviewing!

File: documentation/source/localization.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      
+
+Sometimes it is useful to initialize the map interactively, i.e. start the robot, look at the measurements and try a few poses until the raw measurment is finally aligned with the map.
+
+.. warning::
+    While you are initializing the pose interactively, both odom to sensor frame and map to odom frame transformations can jump!
+
+
+
+You caninitialize the pose interactively with the 2D pose estimate tool in Rviz which is shown in the video below:
```
Comment (tutunarsl) [inline documentation/source/localization.rst:33]: Minor typo
```suggestion
You can initialize the pose interactively with the 2D pose estimate tool in Rviz, which is shown in the video below:
```

File: documentation/source/localization.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+You can do this by calling the Trigger service "initialize_slam". After that, the mapping will start and the scans will be
+inserted into the map if ``is_merge_scans_into_map`` is set to true.
+
+
+
+Alternatively, you can use the 3d interactive marker which is shown the other video below:
+`initialize with interactive marker <https://youtu.be/ePI3SvR3zpw>`_ . 
+
+You have to enable the interactive marker first. Then you can drag the marker to
+the estimated robot pose and then right click on the marker and select "Set pose". If the icp diverges, try better aligning the marker and doing it again. As you drag the marker, the blue pointcloud shows what the raw measurment
```
Comment (tutunarsl) [inline documentation/source/localization.rst:46]: Minor typo
```suggestion
the estimated robot pose, and then right click on the marker and select "Set pose". If the icp diverges, try better aligning the marker and doing it again. As you drag the marker, the blue pointcloud shows what the raw measurement
```

File: documentation/source/localization.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+would look like if the robot was at that given pose. Essentially, the blue pointcloud should fit nicely against the map. Once you have it aligned nicely,
+right click and select "initialize slam" which will star the mapping. Scans will be
```
Comment (tutunarsl) [inline documentation/source/localization.rst:48]: Minor typo
```suggestion
right click and select "initialize slam" which will start the mapping. Scans will be
```

File: open3d_slam/src/SubmapCollection.cpp [hunk 4]
```diff
@@ -253,14 +253,17 @@ const Constraints& SubmapCollection::getOdometryConstraints() const {
...
+		const TimestampedSubmapIds &loopClosureCandidatesIdxs)  {
 	Constraints retVal;
 	for (const auto &id : loopClosureCandidatesIdxs) {
 		const auto constraints = placeRecognition_.buildLoopClosureConstraints(mapToRangeSensor_, *this,
 				adjacencyMatrix_, id.submapId_, activeSubmapIdx_, id.time_);
 		if (!constraints.empty()) {
 			for (int i = 0; i < constraints.size(); ++i) {
 				retVal.push_back(constraints.at(i));
+				const int v1 = constraints.at(i).sourceSubmapIdx_;
+				const int v2 = constraints.at(i).sourceSubmapIdx_;
```
Comment (tutunarsl) [inline open3d_slam/src/SubmapCollection.cpp:265]: Does this constraint add a `self` edge in the loop closure graph? , from constraint `i` to again `i` 

Sorry for the naive question, I am rusty on loop closure

**Review by tutunarsl — COMMENTED** [review]
Nice documentation definitely makes it easy and intuitive to follow. Dropped a few minor typo fixes and a question

File: open3d_slam/src/SubmapCollection.cpp (hunk 4 above)
Comment (jelavice) [reply open3d_slam/src/SubmapCollection.cpp:265]: Good catch! This is definitely a bug. The way it is implemented now, it does not affect anything, however that was not the intention. I will fix this in a separate PR.

**Review by tutunarsl — APPROVED** [review]
LGTM, concise and nice PR

**Pull request merged**